### PR TITLE
Allow list, tuple and None in to_cpu and to_gpu

### PR DESCRIPTION
**Pull request by niboshi**
This PR improves `cuda.to_cpu` and `cuda.to_gpu` by accepting `None` and a sequence of arrays.

### CODE REVIEW

File: chainer/cuda.py [hunk 1]
```diff
@@ -237,16 +237,17 @@ def to_gpu(array, device=None, stream=None):
     """Copies the given CPU array to the specified device.
 
     Args:
-        array: Array to be sent to GPU.
+        array (numpy.ndarray, cupy.ndarray, list or tuple): Array or arrays to
```
Comment (unnonouno) [inline chainer/cuda.py]: Do we need to write that `None` is supported?

File: tests/chainer_tests/test_cuda.py [hunk 2]
```diff
@@ -194,6 +194,40 @@ def test_cupy_array_async2(self):
...
+        d = cuda.cupy.random.uniform(-1, 1, (2, 2))
+        xs = typ([a, b, c, d, None, a, b, None, c, d])
+        xs_cpu = cuda.to_cpu(xs)
+
+        assert isinstance(xs_cpu, typ)
+        assert len(xs) == len(xs_cpu)
+        for i in (0, 1, 2, 3, 5, 6, 8, 9):
+            assert isinstance(xs_cpu[i], numpy.ndarray)
+            cuda.cupy.testing.assert_array_equal(xs[i], xs_cpu[i])
+        assert xs_cpu[0] is a
```
Comment (unnonouno) [inline tests/chainer_tests/test_cuda.py]: Do we need to use `assertIs`?

File: chainer/cuda.py (hunk 1 above)
Comment (niboshi) [reply chainer/cuda.py]: Fixed.

File: tests/chainer_tests/test_cuda.py (hunk 2 above)
Comment (niboshi) [reply tests/chainer_tests/test_cuda.py]: I think `assert` is better, becase it's more straight-forward.
This is one of the benefits we discussed when we moved to PyTest.

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by niboshi** [comment]
Fixed the test error.

**Comment by unnonouno** [comment]
LGTM

**Pull request merged**